src/field.js

```diff
@@ -1,14 +1,18 @@
 class Field {
   constructor(type, options = {}) {
-    this.name = ""
+    this.name = ''
     this.type = type
     this.options = options
+    this.isId = !!options.isId
     this._validations = null
+
+    if (this.isId)
+      if (!(this.type === Number) && !(this.type === String)) this.isId = false
```

Comment:
I don't think it should have any constrain for the type of the ID at domain level. Let the infrastructure glue or layer decides if such restriction should be applied.